src/controller.rs

```diff
@@ -150,13 +198,52 @@ async fn launch_job(
             backoff_limit: Some(1),
             template: PodTemplateSpec {
                 spec: Some(PodSpec {
+                    volumes: Some(vec![
+                        Volume {
+                            name: "credentials".to_string(),
+                            secret: Some(SecretVolumeSource {
+                                secret_name: Some("gar-docker-secret".to_string()),
+                                items: Some(vec![KeyToPath {
+                                    key: ".dockerconfigjson".to_string(),
+                                    path: "config.json".to_string(),
+                                    ..Default::default()
+                                }]),
+                                ..Default::default()
+                            }),
+                            ..Default::default()
+                        },
+                        Volume {
+                            name: "overlay".to_string(),
+                            config_map: Some(ConfigMapVolumeSource {
+                                name: Some(configmap_name),
+                                ..Default::default()
+                            }),
+                            ..Default::default()
+                        },
+                    ]),
                     restart_policy: Some("Never".to_string()),
                     containers: vec![Container {
-                        name: "kubecfg".to_string(),
+                        name: "kubecfg-render".to_string(),
```

Comment:
For now I think this is okay, once we have the `kubectl apply` step, I believe this can move to an `initContainer` instead and we can use that output which is written to an `emptyDir` for the apply portion.

I think that is an easy reshuffle to do.